WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt

```diff
@@ -123,14 +123,18 @@ class JetpackMigrationViewModel @Inject constructor(
             }
 
             migrationState is Initial -> emit(Loading)
+
             migrationState is Migrating
-                    || migrationState is Successful && !continueClicked -> emit(
-                initWelcomeScreenUi(migrationState.data, continueClicked)
-            )
+                    || migrationState is Successful && !continueClicked && migrationState.data.sites.isNotEmpty() ->
+                emit(initWelcomeScreenUi(migrationState.data, continueClicked))
 
             migrationState is Successful -> when {
-                !notificationContinueClicked -> emit(initNotificationsScreenUi())
-                else -> emit(initSuccessScreenUi())
+                !notificationContinueClicked && migrationState.data.sites.isNotEmpty() ->
+                    emit(initNotificationsScreenUi())
+                else -> {
+                    if (migrationState.data.sites.isNotEmpty()) emit(initSuccessScreenUi())
+                    else emit(initSuccessScreenUiForNoSites())
```

Comment:
TBH, I felt a little apprehensive in this area of the code because I didn't feel like I had a good grasp on the complexities of the migration state. With that said, I went ahead and took your suggestion, tested it, and added it with commit.

Reply:
> I didn't feel like I had a good grasp on the complexities of the migration state

The "gist" of it is that the `migrationState` alone isn't enough to drive the `uiState`, since user input (i.e. tapping buttons, etc.) affects when the next screens are shown. The `combineTransform` _mostly_ transforms a given `migrationState` into a `uiState`, but also includes these state of those user interactions so that progress through the flow (from a UI perspective) can be governed by both. E.g. tapping continue before the database is fully migrated will show a pending state before success (this is not often encountered when testing small sites, because the migration can be too fast).

In retrospect, at least _some_ of what is added here (specifically, I'm thinking of the "showDeleteState") should probably been it's own activity 🤷‍♂️ . We could have reused code via composition (especially that we are using compose), so it feels more fitting that it would have been a separate concern.